libazureinit/src/config.rs

```diff
@@ -274,6 +287,106 @@ impl Default for AzureInitLogPath {
     }
 }
 
+/// Runcmd argument struct.
+///
+/// Each command is specified as an array of string arguments,
+/// passed directly to the underlying executable. For example:
+/// ```yaml
+/// runcmd:
+///   commands:
+///     - ["echo", "hello", "world"]
+///     - ["apt-get", "update"]
+///     - ["bash", "-c", "echo 'Done!' && exit 0"]
+/// ```
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(default)]
+pub struct RunCmd {
+    /// A list of commands, where each command itself is a list of arguments.
+    /// Defaults to an empty list if not specified.
+    pub commands: Vec<Vec<String>>,
+
+    /// Whether provisioning should fail if `runcmd` encounters an error.
+    /// Defaults to `true`.
+    pub fail_provisioning_upon_failure: bool,
+}
+
+impl Default for RunCmd {
+    fn default() -> Self {
+        Self {
+            commands: Vec::new(),
+            fail_provisioning_upon_failure: true,
+        }
+    }
+}
+
+// Represents an instruction to write a file during provisioning.
+///
+/// # Fields
+/// - `path`: The destination path of the file.
+/// - `owner`: The username (e.g. "root") that will own the file. Defaults to `root`.`
+/// - `group`: The group name (e.g. "root") that will own the file. Defaults to `root`.
+/// - `permissions`: Numeric file permissions in octal (e.g. 0o600). Defaults to `o600`.
+/// - `content`: Raw file content as a string (e.g., `"hello"` or base64-encoded).
+/// - `encoding`: Optional encoding for the content. Supported values:
+///     - `"b64"`: Base64-encoded content.
+///     - `"b64+gz"`: Base64-encoded and gzipped content.
```

Comment:
This should be `gz+b64` to be consistent with typical usage (as in, the content shall be gzipped, then base64 encoded).